homeassistant/components/habitica/sensor.py

```diff
@@ -263,25 +207,30 @@ def __init__(
             identifiers={(DOMAIN, entry.unique_id)},
         )
 
-    async def async_update(self) -> None:
-        """Update Sensor state."""
-        await self.coordinator.update()
-        data = self.coordinator.data
+    @property
+    def native_value(self) -> StateType:
+        """Return the state of the device."""
+        data = self.coordinator.data.user
         for element in self.entity_description.value_path:
             data = data[element]
-        self._attr_native_value = data
+        if isinstance(data, (str, int, float)):
+            return data
+        return None
```

Comment:
Why do we do this?

Reply:
linter complained that dict is not allowed as return value

Reply:
🤷‍♂️
```
homeassistant/components/habitica/sensor.py:214: error: Incompatible return value type (got "dict[Any, Any]", expected "str | int | float | None")  [return-value]
```